### DISCUSSION TITLE
Too many requests in form with many relationship pickers

### DISCUSSION THREAD

**Discussion by zedascouves** [Budibase/budibase#11203] [Help]
Hello,

I'm building a form to populate a table with many (32) foreign keys (28 of them point to the same foreign table, which is basically an extended boolean table: true, false, not applicable), and I'm using relationship pickers for those fields. I'm getting this error message whenever that page loads:

![image](https://github.com/Budibase/budibase/assets/594831/2be4ba9c-f69b-49f1-a48d-5156c6684769)

I'm currently trying to divide this in a two step form, but is this a good way to avoid this error? What other options would I have?

Any help is appreciated!

**Comment by melohagan** [comment]
Hey @zedascouves 

Are you cloud-hosted or self-hosted?

**Comment by zedascouves** [comment]
Hi @melohagan,

I'm cloud-hosted.

The 2-step form seems to have worked though.

**Reply by melohagan** [reply]
Could you share a screenshot of your **Relationships** tab for your table please?

**Reply by zedascouves** [reply]
I'd love to, but, for some reason, when I click on that tab, the whole page freezes and I need to refresh the page to be able to do anything again.

![imagem](https://github.com/Budibase/budibase/assets/594831/3971e877-4f05-4b66-89ce-572b3f69c0a9)

This started happening when I fetched only some tables and it removed all the other ones. This probably messed up the existing relationships, even after fetching everything back again. I also had to recreate all the relationships.

In any case, the original error (too many requests) started happening before this.

**Reply by melohagan** [reply]
Could you open the console log and share any errors you see please? On Chrome that'd be Fn+F12

**Reply by zedascouves** [reply]
Sure:

> Uncaught (in promise) TypeError: h1 is undefined
>     l1 https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1640
>     l1 https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1640
>     update https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1640
>     init$2 https://valis.budibase.app/builder/assets/index.b86c6c8f.js:4
>     Relationships https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1640
>     create_if_block_2$10 https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1643
>     p https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1643
>     update_slot_base https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1
>     p https://valis.budibase.app/builder/assets/index.b86c6c8f.js:34
>     update https://valis.budibase.app/builder/assets/index.b86c6c8f.js:4
>     flush https://valis.budibase.app/builder/assets/index.b86c6c8f.js:4
>     promise callback*schedule_update https://valis.budibase.app/builder/assets/index.b86c6c8f.js:4
>     make_dirty https://valis.budibase.app/builder/assets/index.b86c6c8f.js:4
>     ctx https://valis.budibase.app/builder/assets/index.b86c6c8f.js:4
>     8g https://valis.budibase.app/builder/assets/index.b86c6c8f.js:249
>     o1 https://valis.budibase.app/builder/assets/index.b86c6c8f.js:4
>     set_store_value https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1
>     m1 https://valis.budibase.app/builder/assets/index.b86c6c8f.js:249
> index.b86c6c8f.js:1640:18739

Cheers!

**Reply by melohagan** [reply]
Hmm, could you also click on this link `https://valis.budibase.app/builder/assets/index.b86c6c8f.js:1640` within the console tab and screenshot the code block it takes you to?

The code is minified when deployed so I need to see some code to figure out whereabouts in the codebase I need to look!

**Reply by zedascouves** [reply]
Well, that was hard. Every time I did this, it crashed my browser (Firefox 115.0.2).

<img width="661" alt="Captura de ecrã 2023-07-14 152005" src="https://github.com/Budibase/budibase/assets/594831/8a7e1b65-c69d-45a8-8c72-9f7634ce1343">
